weedcoco/repo/deposit.py

```diff
@@ -35,22 +36,26 @@ def get_all_existing_hash(repository_dir):
     }
 
 
-def setup_dataset_dir(repository_dir):
-    if not os.path.isdir(repository_dir):
-        os.mkdir(repository_dir)
-        dataset_dir = repository_dir / "dataset_1"
+def setup_dataset_dir(repository_dir, upload_id=None):
+    if upload_id is None:
+        if not os.path.isdir(repository_dir):
+            os.mkdir(repository_dir)
```

Comment:
Avoid race conditions:
```suggestion
        try:
            os.mkdir(repository_dir)
        except FileExistsError as e:
            if not repository_dir.is_dir():
                raise RuntimeError(f"{repository_dir} exists but is not a directory") from e
```